### DISCUSSION TITLE
CRUD Service: how to handle two upsert-one on the same document at the same time

### DISCUSSION THREAD

**Discussion by SalvadorCali** [mia-platform/community#353] [Q&A] [answered]
### Which section?

Marketplace

### Which is the version?

CRUD Service 5.0.1

### Description

Hi everyone!
One of our clients has the following flow:
- **service A** calls **service B** two times with different parameters but the same `requestId`
- for each request, **service B**:
    - calls an external endpoint
    - calls the `upsert-one` of the `crud-service` to store the response on Mongo

The goal is to create on Mongo a single document with the received `requestId` with data received from the two calls, but:
- the result of the first `upsert-one` is 200 and the document is correcly inserted/updated
- the result of the second `upsert-one` is 500 and the document isn't inserted/updated

Is there a suggested way to handle this flow with the `crud-service`?

**Comment by danibix95** [comment, marked as answer]
Hi @SalvadorCali, I tried calling multiple times the `/upsert-one` endpoint over a collection and the result was the expected one. I've been using the latest version of CRUD Service.

Have you tried upgrading the service version to `v6.9.4`?  
Furthermore, I would recommend to verify whether any unique index is set and what they cover.

**Reply by SalvadorCali** [reply]
Ok @danibix95, thanks. 
We'll try to update the `crud-service` version and we'll update you.

**Reply by SalvadorCali** [reply]
I confirm that the update to version `v6.9.4` fixed the problem.
